### PR TITLE
Refactor expo-image-picker on iOS

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

- Follow up #30840 (removed dead code after minimum iOS version bump)
- Existing code is a bit messy, so I'm trying to improve it by moving and refactoring some stuff and converting some callbacks to async/awaits
- Replaced the deprecated `kUTTypeImage` and `kUTTypeMovie` with the new `UTType`
- Made it resolve earlier (faster) by removing unnecessary dispatches on the main thread and not waiting for the view controller to be entirely dismissed
- Preparation to make the returned assets shared refs

I also left some additional comments below to make it easier to review

# Test Plan

Tested in test-suite, NCL and https://github.com/enterteg/SlowImagePicker on simulator and real device

### CODE REVIEW

File: packages/expo-image-picker/ios/ImagePickerHandler.swift [hunk 1]
```diff
@@ -54,35 +52,25 @@ internal class ImagePickerHandler: NSObject,
...
-    }
+    handlePickingCancellation()
+    picker.dismiss(animated: true)
   }
 
   // MARK: - PHPickerViewControllerDelegate
 
-  @available(iOS 14, *)
   func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
-    DispatchQueue.main.async {
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/ImagePickerHandler.swift:76]: All these delegate functions are already invoked on the main thread – no reason to delay the execution.

File: packages/expo-image-picker/ios/ImagePickerHandler.swift [hunk 2]
```diff
@@ -54,35 +52,25 @@ internal class ImagePickerHandler: NSObject,
...
-      picker.dismiss(animated: true) { [weak self] in
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/ImagePickerHandler.swift:77]: Probably it doesn't make sense to wait until the picker is completely dismissed – better the handle these assets right away.

File: packages/expo-image-picker/ios/ImagePickerOptions.swift [hunk 3]
```diff
@@ -18,7 +17,7 @@ internal struct ImagePickerOptions: Record {
   var aspect: [Double]
 
   @Field
-  var quality: Double?
+  var quality: Double = 1.0
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/ImagePickerOptions.swift:20]: This is a big change, but I think this is what people usually expect – to keep the original image.

File: packages/expo-image-picker/ios/ImagePickerUtils.swift
Comment (tsapeta) [inline packages/expo-image-picker/ios/ImagePickerUtils.swift:1]: These functions are also used in the modules core so I'm considering making it a public API, under `AsyncUtils` scope or something like that.

File: packages/expo-image-picker/ios/ImageUtils.swift [hunk 4]
```diff
@@ -0,0 +1,324 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+
+import Photos
+import UniformTypeIdentifiers
+import ExpoModulesCore
+
+internal struct ImageUtils {
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/ImageUtils.swift:7]: It's mostly just moved out from `MediaHandler`

File: packages/expo-image-picker/ios/VideoUtils.swift [hunk 5]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+
+import AVFoundation
+
+internal struct VideoUtils {
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/VideoUtils.swift:5]: Also moved from `MediaHandler`

**Review by lukmccall — APPROVED** [review]

**Review by alanjhughes — APPROVED** [review]

**Comment by hadnet** [comment]
is the dark mode theme working in the expo-image-picker on iOS? I'm using Expo Go and the expo-image-picker and it does not recognize the phone dark mode theme. The image picker screen is always light theme.

**Pull request merged**